SignalServiceKit/src/Loki/Protocol/Meta/SessionMetaProtocol.swift

```diff
@@ -58,9 +58,16 @@ public final class SessionMetaProtocol : NSObject {
     @objc(isThreadNoteToSelf:)
     public static func isThreadNoteToSelf(_ thread: TSThread) -> Bool {
         guard let thread = thread as? TSContactThread else { return false }
+        let hexEncodedPublicKey = thread.contactIdentifier()
         var isNoteToSelf = false
         storage.dbReadConnection.read { transaction in
-            isNoteToSelf = LokiDatabaseUtilities.isUserLinkedDevice(thread.contactIdentifier(), transaction: transaction)
+            //When session reset status is not none, only send the message to linked device
+            let masterHexEncodedPublicKey = storage.getMasterHexEncodedPublicKey(for: hexEncodedPublicKey, in: transaction) ?? hexEncodedPublicKey
+            if let masterThread = TSContactThread.getWithContactId(masterHexEncodedPublicKey, transaction: transaction), masterThread.sessionResetStatus == .none {
```

Comment:
Mixing in session reset logic here is bad. This method should just tell you whether the given thread is a note to self. If at the call site the session reset logic matters you should probably just handle that case manually.

Reply:
Yeah, I was thinking of that when doing this on Android. It is confusing to put the logic here.